src/app/tx/Form.tsx

```diff
@@ -0,0 +1,43 @@
+'use client';
+import { usePathname, useRouter } from 'next/navigation';
+import { Tooltip } from 'react-tooltip';
+import { isValidTxHash } from '@/utils';
+
+import 'react-tooltip/dist/react-tooltip.css';
+
+const Form = () => {
+  const pathname = usePathname();
+  const router = useRouter();
+  const tx = pathname.split('/tx/')[1];
+
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+    const form = e.target as HTMLFormElement;
+    const formData = new FormData(form);
+    const txValue = formData.get('txInput')?.toString();
+
+    if (isValidTxHash(txValue)) {
+      router.replace(`/tx/${txValue}`);
+    } else {
+      form.reset();
+      router.replace('/tx');
+    }
+  };
+
+  return (
+    <>
+      <form className="form-container" onSubmit={handleSubmit}>
+        <input
+          name="txInput"
+          placeholder="Paste your transaction hash"
+          className="input-style"
+          defaultValue={tx}
```

Comment:
We should disable this input if the user is already connected with their wallet as it doesn't let us change the filled wallet address anyway.

Reply:
^ We can use `wagmi` for checking that